client.go

```diff
@@ -505,10 +506,21 @@ func (s *Client) LoopInQuote(ctx context.Context,
 		}, nil
 	}
 
-	// Get estimate for miner fee.
+	// Get estimate for miner fee. If estimating the miner fee for the
+	// requested amount is not possible because lnd's wallet cannot
+	// construct a sample TX, we just return zero instead of failing the
+	// quote. The user interface should inform the user that fee estimation
+	// was not possible.
 	minerFee, err := s.lndServices.Client.EstimateFeeToP2WSH(
 		ctx, request.Amount, request.HtlcConfTarget,
 	)
+	if err != nil && strings.Contains(err.Error(), "insufficient funds") {
```

Comment:
String matching? Is there no other possibility?

Reply:
I don't see any other possibility. We get this error over a gRPC boundary so we can't really type match anything. Also the error itself is not exported in `btcsuite/btcwallet/wallet/txauthor/author.go` so we can't even compare the message string directly.

Reply:
It would require an explicit error code to be added to `lnrpc.proto`. Or use one of the http status codes. That doesn't require a proto change.

Reply:
Discussed offline. Change requires lnd change, so if we want to move forward now, string matching is the only option.

Reply:
Could add a TODO there

Reply:
Okay, added a TODO.